File: CloudFlow/FrontEnd-App/src/app/shared/model/calendar/calendar-user-rights.model.ts
import { OrganizationEntityModel } from "../organization-entity.model";
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class CalendarUserRightsModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("calendarId", Number)
	public calendarId: number = null;

	@JsonProperty("user", OrganizationEntityModel)
	public user: OrganizationEntityModel = null;

	@JsonProperty("view", Boolean)
	public view: boolean = false;

	@JsonProperty("add", Boolean)
	public add: boolean = false;

	@JsonProperty("edit", Boolean)
	public edit: boolean = false;

	@JsonProperty("delete", Boolean)
	public delete: boolean = false;
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/calendar/CalendarUserRightsModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.calendar;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.OrganizationEntityModel;

public class CalendarUserRightsModel {

	private Long id;
	private Long calendarId;
	private OrganizationEntityModel user;
	private boolean view;
	private boolean add;
	private boolean edit;
	private boolean delete;
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public Long getCalendarId() {
		return calendarId;
	}
	public void setCalendarId(Long calendarId) {
		this.calendarId = calendarId;
	}
	public OrganizationEntityModel getUser() {
		return user;
	}
	public void setUser(OrganizationEntityModel user) {
		this.user = user;
	}
	public boolean isView() {
		return view;
	}
	public void setView(boolean view) {
		this.view = view;
	}
	public boolean isAdd() {
		return add;
	}
	public void setAdd(boolean add) {
		this.add = add;
	}
	public boolean isEdit() {
		return edit;
	}
	public void setEdit(boolean edit) {
		this.edit = edit;
	}
	public boolean isDelete() {
		return delete;
	}
	public void setDelete(boolean delete) {
		this.delete = delete;
	}
}
